pkg/app/piped/executor/analysis/analyzer.go

```diff
@@ -63,15 +67,18 @@ func (a *analyzer) run(ctx context.Context) error {
 		select {
 		case <-ticker.C:
 			expected, reason, err := a.evaluate(ctx, a.query)
-			if errors.Is(err, context.DeadlineExceeded) && ctx.Err() == context.DeadlineExceeded {
+			switch {
+			case errors.Is(err, context.DeadlineExceeded) && ctx.Err() == context.DeadlineExceeded:
```

Comment:
nits, instead of switch case statement, I think simplify it by using `if errors.Is(err, context.DeadlineExceeded) && ctx.Err() == context.DeadlineExceeded` and return early may make it more readable 🤔  But it's up to you 😁

Reply:
Thank you. Due to https://github.com/pipe-cd/pipe/pull/1771#discussion_r600188374, I could make up my mind to do as you suggested.